test/fixtures/controller.line/backgroundColor/scriptable.js

```diff
@@ -6,18 +6,19 @@ module.exports = {
 			datasets: [
 				{
 					// option in dataset
-					data: [0, 5, 10, null, -10, -5],
-					pointBackgroundColor: function(ctx) {
-						var value = ctx.dataset.data[ctx.dataIndex] || 0;
-						return value > 8 ? '#ff0000'
-							: value > 0 ? '#00ff00'
-							: value > -8 ? '#0000ff'
-							: '#ff00ff';
+					data: [4, 5, 10, null, -10, -5],
+					backgroundColor: function(ctx) {
+						var dataTotal = ctx.dataset.data.reduce(function(a, b) {
+							return a + b;
+						});
+						var c = dataTotal >= 0 ? '#D60000' : '#44DE28';
```

Comment:
We should test that we receive the correct `dataIndex` in case we are resolving point options, maybe something like:

```javascript
var index = (ctx.dataIndex === undefined ? ctx.dataIndex : ctx.datasetIndex);
return index === 0 ? '#ff0000'
  : index === 1 ? '#00ff00'
  : index === 2 ? '#0000ff'
  : '#ff00ff'
```

Using `reduce()` makes the test a bit more complex to read and I would use `ff` / `00` colors for a better image diff. If that makes sense, I would use this approach instead of `reduce()` in all these new tests.

Reply:
@simonbrunel I think you meant: var index = (ctx.dataIndex !== undefined ? ctx.dataIndex : ctx.datasetIndex) ?

Reply:
Yes, sorry. I fixed my previous comment.